### PR TITLE
Added allowNewEnroll on SmartSelfie, Biometric, DocV and EnhancedDocV

### PR DESCRIPTION
**Pull request by jumaallan**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

Added allowNewEnroll on SmartSelfie, Biometric, DocV and EnhancedDocV

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

### CODE REVIEW

File: Example/Podfile.lock [hunk 1]
```diff
@@ -28,4 +28,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: b024c66547f30afaee4b2f86d064c8edcba8cbef
 
-COCOAPODS: 1.14.3
+COCOAPODS: 1.12.1
```
Comment (beastawakens) [inline Example/Podfile.lock:31]: Is this downgrade intentional?

File: Example/SmileID/HomeViewModel.swift [hunk 2]
```diff
@@ -18,7 +18,7 @@ class HomeViewModel: ObservableObject,
 
     init(config: Config) {
         partnerId = config.partnerId
-        SmileID.initialize(config: config, useSandbox: SmileID.useSandbox)
+        SmileID.initialize(config: config, useSandbox: true)
```
Comment (beastawakens) [inline Example/SmileID/HomeViewModel.swift:21]: Is hardcoding this to be true always also intentional?

File: Sources/SmileID/Classes/Networking/Models/PrepUpload.swift [hunk 3]
```diff
@@ -4,38 +4,37 @@ public struct PrepUploadRequest: Codable {
...
-    /// I've also made this false until we have this properly
-    /// documented and done on both android and iOS
-    public var allowNewEnroll = "false"
     public var useEnrolledImage = false
     public var retry = "false" /// backend is broken needs these as strings
 
     public init(
         partnerParams: PartnerParams,
-        callbackUrl: String? = "",
+        callbackUrl: String? = SmileID.callbackUrl,
```
Comment (beastawakens) [inline Sources/SmileID/Classes/Networking/Models/PrepUpload.swift:19]: I know this also needs to be fixed but do you want this change in this PR or a separate one?

File: Sources/SmileID/Classes/Networking/Models/PrepUpload.swift [hunk 4]
```diff
@@ -4,38 +4,37 @@ public struct PrepUploadRequest: Codable {
...
         timestamp: String = String(Date().millisecondsSince1970),
         signature: String = "",
-        allowNewEnroll: String = "false",
         useEnrolledImage: Bool = false,
         retry: String = "false"
     ) {
         self.partnerParams = partnerParams
         self.callbackUrl = callbackUrl
+        self.allowNewEnroll = allowNewEnroll
+        self.partnerId = partnerId
```
Comment (beastawakens) [inline Sources/SmileID/Classes/Networking/Models/PrepUpload.swift:32]: Should this be duplicated? with L33...

**Review by beastawakens — CHANGES_REQUESTED** [review]
Some comments/questions

File: Example/Podfile.lock (hunk 1 above)
Comment (jumaallan) [reply Example/Podfile.lock:31]: Good catch, let me update my cocoapods

File: Sources/SmileID/Classes/Networking/Models/PrepUpload.swift (hunk 4 above)
Comment (jumaallan) [reply Sources/SmileID/Classes/Networking/Models/PrepUpload.swift:32]: Good catch - thanks

Fixed

File: Example/SmileID/HomeViewModel.swift (hunk 2 above)
Comment (jumaallan) [reply Example/SmileID/HomeViewModel.swift:21]: This is the sample app, so I was defaulting it to use Sandbox. This matches the exact setup on android

Comment (beastawakens) [reply Example/SmileID/HomeViewModel.swift:21]: Aaah, gotcha!

File: Sources/SmileID/Classes/Networking/Models/PrepUpload.swift (hunk 3 above)
Comment (jumaallan) [reply Sources/SmileID/Classes/Networking/Models/PrepUpload.swift:19]: Here is fine. I will make sure I separate PRs in the future

**Review by beastawakens — APPROVED** [review]
With the proviso that it's been a hot minute since I've done any iOS stuff, this looks good to me! 🙌🏼

**Review by beaglebets — APPROVED** [review]
